src/monattack.cpp

```diff
@@ -221,6 +223,71 @@ void mattack::shriek(monster *z, int index)
     sounds::sound(z->pos(), 50, _("a terrible shriek!"));
 }
 
+void mattack::shriek_alert(monster *z, int index)
+{
+    if( !z->can_act() || z->has_effect("shrieking")) {
+        return;
+    }
+
+    Creature *target = z->attack_target();
+
+    if(g->u.sees( *z )){
+    add_msg( _("The %s begins shrieking!"), z->name().c_str());
```

Comment:
Shouldn't this be below the next `if` clause? Currently one may get this message, but the monster may actually not shriek at all (because target is null or too far away or invisible).